index.js

```diff
@@ -17,7 +19,13 @@ module.exports = function (str, filepath) {
 
 	return pify(mkdirp, Promise)(path.dirname(fullpath))
 		.then(function () {
-			return pify(fs.writeFile, Promise)(fullpath, str);
+			return isStream(str) ?
+				new Promise((resolve, reject) => {
+					str.pipe(fs.createWriteStream(fullpath))
```

Comment:
This should already be [doing that](https://nodejs.org/api/stream.html#stream_readable_pipe_destination_options).

> The readable.pipe() method returns a reference to the destination stream ...

The `.on('error', reject)` below is handling errors on the writeable stream.

<hr>

As for the code style. Like this?

```js
const pipeStream = () => {
    return new Promise((resolve, reject) => {
        str.pipe(fs.createWriteStream(fullpath))
            .on('error', reject)
            .on('finish', resolve);
    });
};

return isStream(str) ? pipeStream() : pify(fs.writeFile, Promise)(fullpath, str);
```